src/python/turicreate/_cython/cy_sarray.pyx

```diff
@@ -525,3 +529,20 @@ cdef class UnitySArrayProxy:
         with nogil:
             proxy = self.thisptr.to_const(val, datatype)
         return create_proxy_wrapper_from_existing_proxy(proxy)
+
+    def __reduce__(self):
+
```

Comment:
I think we need to call `self.materialize()` here.

Reply:
The save does that internally, so it should be fine.